lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosValidation.cxx

```diff
@@ -0,0 +1,187 @@
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <algorithm>
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/FunctionalChaosValidation.hxx"
+#include "openturns/Os.hxx"
+#include "openturns/OSS.hxx"
+#include "openturns/DesignProxy.hxx"
+#include "openturns/LeastSquaresMethod.hxx"
+#include "openturns/KFoldSplitter.hxx"
+#include "openturns/MetaModelValidation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(FunctionalChaosValidation)
+
+static const Factory<FunctionalChaosValidation> Factory_FunctionalChaosValidation;
+
+/* Default constructor */
+FunctionalChaosValidation::FunctionalChaosValidation()
+  : MetaModelValidation()
+{
+  // Nothing to do
+}
+
+/* Default constructor */
+
+FunctionalChaosValidation::FunctionalChaosValidation(const FunctionalChaosResult & functionalChaosResult,
+                                                     const CrossValidationMethod cvMethod, 
+                                                     const UnsignedInteger & kParameter)
+  : MetaModelValidation(functionalChaosResult.getOutputSample()
+      , ComputeMetamodelCrossValidationPredictions(functionalChaosResult, cvMethod, kParameter))
+  , functionalChaosResult_(functionalChaosResult)
+{
+  if ((cvMethod != LEAVEONEOUT) && (cvMethod != KFOLD))
+    throw InvalidArgumentException(HERE) << "The method " << cvMethod << " is not available.";
+  cvMethod_ = cvMethod;
+  if (kParameter < 1)
+    throw InvalidArgumentException(HERE) << "Cannot set k parameter of K-Fold method to " << kParameter << " which is lower than 1";
+  const UnsignedInteger sampleSize = functionalChaosResult_.getInputSample().getSize();
+  if (kParameter > sampleSize)
+    throw InvalidArgumentException(HERE) << "Cannot set k parameter of K-Fold method to " << kParameter 
+      << " which is larger than the sample size =" << sampleSize;
+  kParameter_ = kParameter;
+  if (ResourceMap::GetAsBool("FunctionalChaosValidation-NoModelSelection") && \
+    functionalChaosResult_.isModelSelection())
+    throw InvalidArgumentException(HERE) << "Cannot perform fast cross-validation "
+      << "with a polynomial chaos expansion involving model selection";
+  if (!functionalChaosResult.isLeastSquares())
+    throw InvalidArgumentException(HERE) << "Error: the polynomial chaos expansion was not computed from regression.";  
```

Comment:
from least squares, not regression. There is **no** noise model.

Reply:
OK. Fixed at 0f12534f2.